### ISSUE TITLE
Useful for decorators too

### ISSUE THREAD

**Issue by mortenscheel** [stevebauman/autodoc-facades#3]
Hi, I wonder if it wouldn't be possible to extend this package to also be able to generate docblocks on classes that use the decorator pattern (like using the ForwardsCalls trait). I suppose the only real differences are that (1) they don't extend a common class like Facade, and (2) their methods are dynamic, not static.
Would you be interested in a PR that adds an additional `autodoc:decorators` command?
Or maybe just a common `autodoc:generate` which uses a publishable config file to determine what to generate.

**Comment by stevebauman** [comment]
Hi @mortenscheel,

This package depends on and uses [Laravel's Facade Documentor](https://github.com/laravel/facade-documenter) tool, so it cannot be extended or modified directly. I believe the implementation would have to be copied over and then modified to get what you're looking for.

Can you share a code example along with docs that should be generated?

**Comment by mortenscheel** [comment]
> I believe the implementation would have to be copied over and then modified to get what you're looking for.

Sorry I thought you were already doing that, since your composer.json publishes vendor/bin/facades.php, but now I see it's just a copy of the original.

I just took a quick look earlier today, and I was able to _almost_ achieve what I wanted by just commenting https://github.com/stevebauman/autodoc-facades/blob/b69c4642e06635418912d5a9a4ceff433b644804/src/Commands/DocumentFacades.php#L66 to allow it to process non Facade classes. As long as the class has a `@see` docblock, the facade-documenter will do its work. The only problem is that the proxied methods are documented as static methods, which isn't the case with decorated classes. By decorated I mean a class that uses the magic `__call()` method to forward calls to unknown methods to another object. Like Eloquent's Model class forwarding calls to Builder, allowing you to call query builder methods directly on a model instance.

You're right, I don't think it's possible to add this feature without modifying the original facades.php, and I see why you wouldn't want to have to maintain that. Anyway just adding a `@mixin` docblock to decorator classes works pretty well too.
